### PR TITLE
Fix for issue 300

### PR DESCRIPTION
**Pull request by steelgate-1**
Fixes issue with onPageCreated not returning valid web_page object.

### CODE REVIEW

File: shim.js [hunk 1]
```diff
@@ -5809,22 +5809,22 @@ dnode.prototype = {};
 function dnode (cons, opts) {
     Stream.call(this);
     var self = this;
-    
+
```
Comment (amir20) [inline shim.js]: Why are all these lines changing?

File: shim.js [hunk 2]
```diff
@@ -5809,22 +5809,22 @@ dnode.prototype = {};
 function dnode (cons, opts) {
```
Comment (amir20) [inline shim.js]: Where is the change to `shim.coffee`

Comment (steelgate-1) [inline shim.js]: The only changes that have been made should be the addition of :

onPageCreated: function(cb){
        page.onPageCreated = function(new_page) {
          var new_page = pageWrap(new_page);
          return cb(new_page);
        };
      },

to shim.js.

I pulled from the latest repo prior to making the commit.  I have not made a change to the shim.coffee as of yet, will make the change and commit momentarily.



Thanks,
Paul Disney | Software Engineer
Steelgate Intelligence Systems
222 Broadway New York, NY 10038
Tel: +1 443 203 8005
Skype: pdisney1
www.steelgateis.com<http://www.steelgateis.com>​

On Sep 16, 2015, at 11:46 AM, Amir Raminfar <notifications@github.com<mailto:notifications@github.com>> wrote:


In shim.js<https://github.com/sgentle/phantomjs-node/pull/310#discussion_r39646898>:

> @@ -5809,22 +5809,22 @@ dnode.prototype = {};
>  function dnode (cons, opts) {


Where is the change to shim.coffee

Comment (amir20) [inline shim.js]: Please make the changes to shim.coffee and then recompile to .js. The diff should be minimal.

**Comment by amir20** [comment]
Change the commit message to `Fixes issue #300`

**Comment by steelgate-1** [comment]
Requested changes have been checked in.


Thanks,
Paul Disney | Software Engineer
Steelgate Intelligence Systems
222 Broadway New York, NY 10038
Tel: +1 443 203 8005
Skype: pdisney1
www.steelgateis.com<http://www.steelgateis.com>​

On Sep 16, 2015, at 11:57 AM, Amir Raminfar <notifications@github.com<mailto:notifications@github.com>> wrote:

Fixes issue #300

File: shim.js (hunk 2 above)
Comment (amir20) [inline shim.js]: Did you recompile the js file? I don't get why the extra lines are showing up. They shouldn't be there if you compiled it.

**Comment by steelgate-1** [comment]
I did compile the file  (coffee -c shim.coffee)  I can try it again, but I’m not sure what the issue is.

On Sep 16, 2015, at 11:57 AM, Amir Raminfar <notifications@github.com<mailto:notifications@github.com>> wrote:


Change the commit message to Fixes issue #300

File: shim.js (hunk 2 above)
Comment (steelgate-1) [inline shim.js]: On compilation of shim.coffee (coffee -c shim.coffee)  the resulting shim.js file is significantly smaller than the current shim.js in the master branch.  I took the compiled part:

onPageCreated: function(cb) {
        if (cb == null) {
          cb = (function() {});
        }
        return page.onPageCreated = function(newpage) {
          return cb(pageWrap(newpage));
        };
      },

 from the resulting shim.js file and pasted it into the pageWrap function leaving the rest of the shim.js file unchanged when compared to the master copy.  On commit the only piece flagged as changed was the above code snippet.  Let me know if you have any questions.


On Sep 16, 2015, at 2:10 PM, Amir Raminfar <notifications@github.com<mailto:notifications@github.com>> wrote:


In shim.js<https://github.com/sgentle/phantomjs-node/pull/310#discussion_r39664911>:

> @@ -5809,22 +5809,22 @@ dnode.prototype = {};
>  function dnode (cons, opts) {


Did you recompile the js file? I don't get why the extra lines are showing up. They shouldn't be there if you compiled it.

**Comment by amir20** [comment]
Merging

**Pull request merged**